yt_dlp/extractor/goplay.py

```diff
@@ -0,0 +1,398 @@
+        # https://github.com/aws/amazon-cognito-identity-js/blob/master/src/AuthenticationHelper.js#L22
+        self.n_hex = 'FFFFFFFFFFFFFFFFC90FDAA22168C234C4C6628B80DC1CD1' + \
+                     '29024E088A67CC74020BBEA63B139B22514A08798E3404DD' + \
+                     'EF9519B3CD3A431B302B0A6DF25F14374FE1356D6D51C245' + \
+                     'E485B576625E7EC6F44C42E9A637ED6B0BFF5CB6F406B7ED' + \
+                     'EE386BFB5A899FA5AE9F24117C4B1FE649286651ECE45B3D' + \
+                     'C2007CB8A163BF0598DA48361C55D39A69163FA8FD24CF5F' + \
+                     '83655D23DCA3AD961C62F356208552BB9ED529077096966D' + \
+                     '670C354E4ABC9804F1746C08CA18217C32905E462E36CE3B' + \
+                     'E39E772C180E86039B2783A2EC07A28FB5C55DF06F4C52C9' + \
+                     'DE2BCBF6955817183995497CEA956AE515D2261898FA0510' + \
+                     '15728E5A8AAAC42DAD33170D04507A33A85521ABDF1CBA64' + \
+                     'ECFB850458DBEF0A8AEA71575D060C7DB3970F85A6E1E4C7' + \
+                     'ABF5AE8CDB0933D71E8C94E04A25619DCEE3D2261AD2EE6B' + \
+                     'F12FFA06D98A0864D87602733EC86A64521F2B18177B200C' + \
+                     'BBE117577A615D6C770988C0BAD946E208E24FA074E5AB31' + \
+                     '43DB5BFCE0FD108E4B82D120A93AD2CAFFFFFFFFFFFFFFFF'
+
+        # https://github.com/aws/amazon-cognito-identity-js/blob/master/src/AuthenticationHelper.js#L49
+        self.g_hex = '2'
+        self.info_bits = bytearray('Caldera Derived Key', 'utf-8')
+
+        self.big_n = self.__hex_to_long(self.n_hex)
+        self.g = self.__hex_to_long(self.g_hex)  # pylint: disable=invalid-name
+        self.k = self.__hex_to_long(self.__hex_hash('00' + self.n_hex + '0' + self.g_hex))  # pylint: disable=invalid-name
+        self.small_a_value = self.__generate_random_small_a()
+        self.large_a_value = self.__calculate_a()
+
+    def authenticate(self, username, password):
+        """ Authenticate with a username and password. """
+        # Step 1: First initiate an authentication request
+        auth_data_dict = self.__get_authentication_request(username)
+        auth_data = json.dumps(auth_data_dict).encode("utf-8")
+        auth_headers = {
+            "X-Amz-Target": "AWSCognitoIdentityProviderService.InitiateAuth",
+            "Accept-Encoding": "identity",
+            "Content-Type": "application/x-amz-json-1.1"
+        }
+        auth_response = self.ie._download_webpage(self.url, None, fatal=False, note="Authenticating Part 1", errnote="Invalid username", data=auth_data, headers=auth_headers)
+        if auth_response == False:
+            raise ExtractorError("Invalid username", expected=True)
+        auth_response_json = self.ie._parse_json(auth_response, None)
+        challenge_parameters = auth_response_json.get("ChallengeParameters")
+
+        challenge_name = auth_response_json.get("ChallengeName")
+        if not challenge_name == "PASSWORD_VERIFIER":
```

Comment:
```suggestion
        if challenge_name != "PASSWORD_VERIFIER":
```
This code path seems to have been never tested